Topic4Services/src/userServices/UserServiceImp.java

```diff
@@ -0,0 +1,44 @@
+package userServices;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserServiceImp implements UserService {
+
+	private Map<String, User> users;
+
+	public UserServiceImp() {
+		users = new HashMap<String, User>();
+	}
+
+	public void create(String id, String pass) {
+		User user = new User(id, pass);
+		users.put(id, user);

```

Comment:
This could give you a null pointer exception too. In this case is a good idea to use defensive programing trying to avoid that kind of issues